#AMPure XP beads DNA purification

The Agencourt AMPure XP PCR purification system utilizes Beckman Coulter’s solid-phase reversible immobilization (SPRI) paramagnetic bead technology for high-throughput purification of PCR amplicons. Agencourt AMPure XP utilizes an optimized buffer to selectively bind DNA fragments 200 bp and larger to paramagnetic beads. Excess primers, nucleotides, salts, and enzymes can be removed using a simple washing procedure. The result is a more purified PCR product. 

Agencourt AMPure XP purified products can be used in the following applications:
 * PCR
 * Sequencing
 * Genotyping
 * Fragment Analysis
 * Primer Walking
 * Cloning

##Table of contents
* [Equipment](https://github.com/repseqio/protocols/blob/master/AMPure%20beads%20purification%20protocol.md#equipment)
* [Reagents](https://github.com/repseqio/protocols/blob/master/AMPure%20beads%20purification%20protocol.md#reagents)
* [Workflow](https://github.com/repseqio/protocols/blob/master/AMPure%20beads%20purification%20protocol.md#workflow)
* [Procedure](https://github.com/repseqio/protocols/blob/master/AMPure%20beads%20purification%20protocol.md#procedure)
* [Troubleshooting](https://github.com/repseqio/protocols/blob/master/AMPure%20beads%20purification%20protocol.md#troubleshooting)

##Equipment
* 1.5 ml tubes;
* magnet plate for 1.5 ml tubes;
* set of automatic pipettes ranging from 10 µl and 1 ml.

##Reagents
* AMPure XP beads (Beckman Coulter);
* 70% EtOH;
* Elution buffer - EB (Qiagen).

##Workflow

![AMPure workflow](https://github.com/repseqio/protocols/blob/master/img/AMPure%20XP.png "AMPure XP overview")

**Figure 1.** Workflow for PCR Purification. Picture from [www.beckmancoulter.com](https://www.beckmancoulter.com/wsrportal/techdocs?docname=B37419AA).

The workflow for the PCR purification process is as follows:

1. Add 0.8 μl AMPure XP per 1.0 μl of sample.
2. Bind DNA fragments to paramagnetic beads.
3. Separation of beads + DNA fragments from contaminants.
4. Wash beads + DNA fragments twice with 70% EtOH to remove contaminants.
5. Elute purified DNA fragments from beads.
6. Transfer to new plate.

##Procedure

1| Add 0.8 μl AMPure XP beads (Beckman Coulter) per 1.0 μl of sample. Mix well by pipetting. Incubate sample for 5 minutes at room temperature 
for maximum recovery.

**CRITICAL STEP:** Before adding shake well the Agencourt AMPure XP beads to resuspend any magnetic particles that may have
settled. 

2| Place the tube with the sample into magnet plate for 2 minutes to separate beads from the solution.

**CRITICAL STEP:** Wait for the solution to clear before proceeding to the next step.

3| Aspirate the cleared solution from the reaction plate and discard.

**CRITICAL STEP:** Do not disturb the spot of separated magnetic beads.

4| Add 200 μl 70% EtOH and incubate 30 seconds at room temperature. Aspirate out the ethanol and discard. 

5| Repeat step 4 (optional).

6| Open the lid. Dry the tube with beads 3 minutes at room temperature.

**NOTE:** A dry time is optional to ensure all traces of EtOH are removed. Do not over dry the beads as this will significantly
decrease elution efficiency.

7| Add 40 μl of elution buffer - EB (Qiagen). Mix well by pipetting. Incubate sample for 5 minutes at room temperature.

8| Place the tube with the sample into magnet plate for 2 minutes to separate beads from the solution.

9| Transfer the eluate witn DNA to a new tube.

##Troubleshooting

Troubleshooting  was taken from [www.beckmancoulter.com](https://www.beckmancoulter.com/wsrportal/techdocs?docname=B37419AA).

| Low Yield / Recovery Problems | Possible Solutions |
|:-----------------------------:|:-------------------|
| Bead Loss |  If beads get aspirated into tips during supernatant removal, the nucleic acid bound to these beads will also be lost. Aspirate slowly and remove as much of the first supernatant as possible without disturbing the bead pellet. If beads are aspirated by accident, dispense everything back into the well, allow the beads to resettle before aspirating again. Try aspirating slower or with a finer pipette. Low concentrated samples will be more susceptible to bead loss since there is less sample keeping the beads in place. Low volume samples will be more susceptible to bead loss since the beads may not reach the level of the magnet in the well. |   
| Large Reaction Volume | Large volume reactions can benefit from an extended binding and separation time. Increase binding time to 10 minutes and ensure all beads are separated before removing the supernatant. |   
| Low Elution Volume |  A small elution volume leads to a decrease in recovery. This is because a small amount of elution buffer always stays behind coating the beads. This volume is dependent on the well shape and the amount of beads in the well, so a smaller elution volume will lead to a higher percentage of eluate staying behind.  | 
| Ethanol |  Ethanol must be at least 70%. When diluting 100% ethanol to 70%, ensure that water and ethanol are measured SEPARATELY before combining due to the miscibility of ethanol. Topping off ethanol with water to dilute will lead to a lower concentration than intended. Stock ethanol can also absorb water from the atmosphere over time leading to a lower concentration. Ensure stock ethanol remains tightly capped when not in use. |
| Insufficient Mixing | Mixing thoroughly during the initial bind mix and elution mix is critical. Before elution do not overdry beads - it leads to a decrease in recovery. Incubation times should also be maintained to ensure the nucleic acid has enough time to bind or dissociate with the beads. Vortexing during binding can be inefficient because of the viscosity of the sample.  | 
